src/components/expression/expression-renderer.js

```diff
@@ -310,15 +350,35 @@ genotet.ExpressionRenderer.prototype.drawMatrixCells_ = function() {
       .range(genotet.data.redYellowScale);
   }
   var transformLeft = 0;
-  var transformTop = this.conditionLabelHeight_ + this.profileHeight_;
+  var transformTop = this.profileHeight_;
   if (this.data.options.showGeneLabels) {
-    transformLeft += this.LABEL_MARGIN_;
+    transformLeft += this.LABEL_MARGIN;
   }
   else if (this.data.options.showProfiles) {
     transformLeft += this.DEFAULT_PROFILE_MARGIN;
   }
   if (this.data.options.showConditionLabels) {
-    transformTop += this.LABEL_MARGIN_;
+    transformTop += this.conditionLabelHeight_ +
+      this.LABEL_MARGIN + this.LABEL_DIFFERENCE;
+  }
+
+  if (this.data.clickedCell) {
+    this.data.clickedCell.row = heatmapData.geneNames.indexOf(
+      this.data.clickedCell.geneName
+    );
+    this.data.clickedCell.column = heatmapData.conditionNames.indexOf(
+      this.data.clickedCell.conditionName
+    );
+    if (this.data.clickedCell.row == -1 ||
+      this.data.clickedCell.column == -1
+    ) {
+      this.unhighlightLabelsForClickedCell_();
+      this.signal('cellUnclick');
+    } else if (
+      this.data.clickedCell.row != -1 && this.data.clickedCell.column != -1
```

Comment:
not sth critical but pls try to end lines with operators like "&&", and not brackets "("
```js
    } else if (this.data.clickedCell.row != -1 &&
      this.data.clickedCell.column != -1)
```